src/background.js

```diff
@@ -266,43 +266,20 @@ async function maybeReopenTab (url, tab, request) {
   return {cancel: true};
 }
 
-function getRootDomain(hostname) {
-  let parts = hostname.split(".");
+function getRootDomain(url) {
```

Comment:
maybe a future performance optimization? since there will be many requests sent to the same domain, can we memo-ize the result of this function so we don't call the full parse logic every time?